homeassistant/components/acer_projector/switch.py

```diff
@@ -116,57 +116,36 @@ def _write_read_format(self, msg: str) -> str:
             return match.group(1)
         return STATE_UNKNOWN
 
-    @property
-    def available(self) -> bool:
-        """Return if projector is available."""
-        return self._available
-
-    @property
-    def name(self) -> str:
-        """Return name of the projector."""
-        return self._name
-
-    @property
-    def icon(self) -> str:
-        """Return the icon."""
-        return ICON
-
-    @property
-    def is_on(self) -> bool:
-        """Return if the projector is turned on."""
-        return self._state
-
-    @property
-    def extra_state_attributes(self) -> dict[str, str]:
-        """Return state attributes."""
-        return self._attributes
-
     def update(self) -> None:
         """Get the latest state from the projector."""
         awns = self._write_read_format(CMD_DICT[LAMP])
         if awns == "Lamp 1":
-            self._state = True
-            self._available = True
+            self._attr_is_on = True
+            self._attr_available = True
         elif awns == "Lamp 0":
-            self._state = False
-            self._available = True
+            self._attr_is_on = False
+            self._attr_available = True
         else:
-            self._available = False
+            self._attr_available = False
 
-        for key in self._attributes:
+        if self.extra_state_attributes is None:
+            return
+        attr = {}
+        for key in self.extra_state_attributes:
             msg = CMD_DICT.get(key)
             if msg:
                 awns = self._write_read_format(msg)
-                self._attributes[key] = awns
+                attr[key] = awns
+        self._attr_extra_state_attributes = attr
```

Comment:
In general using class attributes makes the code simple and shorter.
This is a case where I am not sure if we end up in a more complicated code. I think that the previous state is much cleaner. @frenck what do you think?

Reply:
I think this code is incorrect.